src/Workspaces/Remote/Core/ProjectSystem/SourceFileInfo.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Remote.ProjectSystem
+{
+    public record struct SourceFileInfo(string FilePath, IReadOnlyList<string> FolderNames)
+    {
+    }
```

Comment:
```suggestion
    public readonly record struct SourceFileInfo(string FilePath, IReadOnlyList<string> FolderNames);
```

personally, i would prefer ImmutableArray as well.  since this is marshalling, that's fine, and it will mean the IROL isn't backed by a mutable collection.

Reply:
IReadOnlyList was chosen following the RPC best practice document earlier.